### DISCUSSION TITLE
Copy element to unrelated container

### DISCUSSION THREAD

**Discussion by HilkopterBob** [zauberzeug/nicegui#743] [Q&A]
### Question

Hey, 

i have following page:
![image](https://user-images.githubusercontent.com/117028104/231172652-a9bb5824-9ea8-4068-809d-0652eaead57d.png)

i need a way of transferring the expansion from  the presets into the jobs section on a click on the start symbol:
![image](https://user-images.githubusercontent.com/117028104/231173053-6283fdbd-39a0-44b2-ac47-f9bbb72d4c11.png)

i think a custom event with a listener inside the jobs panel would be a good way, but i dont know if custom events are compatible with nicegui? if there is another, easier way, let me know!

**Comment by falkoschindler** [comment]
Sorry, I don't understand what exactly you're asking. Is it about how to copy (or move) elements, as the title suggests? (Similar to https://github.com/zauberzeug/nicegui/discussions/703#discussioncomment-5518667?) Or is it about registering custom events?

"i dont know if custom events are compatible" - Have you tried it? What happens? What does not work?

**Reply by HilkopterBob** [reply]
sorry for the unclear question, i will elaborate more:

on my nicegui page i have two tabviews with two tabs each as shown in the picture above. I want that when you click on the play button of an expansion element (see picture) that an element of the second tab view tab 1 (Jobs, see picture) is created.

Unfortunately, I don't know how to implement this, because my nicegui program is very modularized and I don't like to pass containers over a large number of functions as parameters. Therefore I wanted to ask if there is a better, more elegant solution to solve this problem.

Or in short:
I am looking for a way to create elements within a container that is not stored in a variable if possible, without having to pass the container as a function parameter, for example.

So much for the explanation of my problem.

What I meant by the events was an idea on my part for a kind of solution using an event listener that creates a container whenever a special event occurs (e.g. play button clicked).

**Reply by falkoschindler** [reply]
Ok, that seems to be mostly a Python problem: How to modularize code and organize communication between those modules. There's probably no easy answer. But I'd like to point you to a similar discussion https://github.com/zauberzeug/nicegui/discussions/661.

Maybe a custom event system is a useful architecture (like we implemented for our robot system RoSys: https://github.com/zauberzeug/rosys/blob/main/rosys/event.py).

Or maybe you just have to find the right cut between modules. If there's interaction between a button and a container somewhere else in the UI - and both UI modules should not need to know about each other -, maybe you can introduce some global module like a state machine, event system, or just some data model everyone can bind from.
